---
title: "Using multi-omics to identify bile- and liver- biomarkers that correlate with
  biliary viability of ECD livers"
author: |
  | Jarno Duiker $^1$, Ramon Reilman $^1$, Tai Vo $^1$, Yamila Timmer $^1$
  | $^1$Hanze Groningen
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
  pdf_document:
    number_sections: false
bibliography: library.bib
lang: en
header-includes: \usepackage{float}
---
```{r, echo=FALSE}
suppressWarnings(library(knitr))
```

# Abstract

# Introduction
Liver transplantation is the first treatment option for patients with liver dysfunctionality (@shen2023role), e.g. in patients with liver cirrhosis or acute liver failure. There is a shortage of donor livers that meet required standards (@Arulraj2011), so now expanded criteria donors (ECD)(@Feng2014) are increasingly used, these consist of livers that do not meet the normally required standards and are often extracted from cardiac death donors (DCD) but also include e.g. elderly donors. Using ECD for liver transplantation can provide a solution for the shortage of liver donors, however, there are also risks associated with ECD livers, such as increased chances of graft failure and death especially in recipients < 60 years old and patients with diabetes (@VanIttersum2016). ECD livers are also more susceptible to biliary complications and ischemia-reperfusion injury, which is damage that occurs when an organ lacking oxygen gets reperfused with oxygen again (@shen2023role).

Certain techniques, such as normothermic machine perfusion (NMP) (@Hosgood2011) can reduce the risk of post-transplant complications. NMP attempts to restore and maintain cell metabolism by simulating the natural environment of the liver, by maintaining a temperature of 37 degrees Celsius and providing the organ with the required oxygen and nutrition (@shen2023role). NMP is considered a better alternative to static cold storage (SCS)(@Lee2009), as it reduces the risk of ischemia-reperfusion injury, early allograft dysfunction (EAD), and biliary complications, improving general graft viability (@shen2023role). NMP also allows evaluation of the donor liver pre-transplant, further reducing transplantation risks (@thorne2023bile). 

@thorne2023bile describe how NMP is used for assessing organ viability before liver transplantation. Here, the general viability of the organ is divided into two distinct measurements. First, is hepatocellular viability (@Groen2024), which focuses on the metabolic functionality of the parenchyma, or the 'functional tissue' of the liver. The second measurement focuses on biliary viability, which focuses on the state of the cholangiocytes, the epithelial cells of the bile ducts. Biliary viability is measured by analyzing the composition of the bile, including glucose levels and pH (@Groen2024), but as of today, there is no golden standard (@Dingfelder2022), and measurement criteria differ between centers. Biliary viability, unlike hepatocellular viability, is not yet adopted in all clinical trials that work with ECD donor livers; however, it has been shown that the use of both hepatocellular as well as biliary viability in the selection of ECD donors has led to a lower incidence of post-transplant complications affecting cholangiocytes, also known as cholangiopathies (@thorne2023bile).


To gain a more comprehensive understanding of the molecular changes that contribute to biliary viability, a multi-omics approach (@Hasin2017) is employed. Multi-omics integrates data from various biological levels, such as genomics, transcriptomics, metabolomics, and proteomics, to provide a view of complex biological systems, cellular responses, and disease progression. Specifically, this paper leverages a multi-omics strategy to identify bile and liver-based biomarkers that correlate with biliary viability. Biomarkers include various biomolecules that can serve as identifiers for a condition, such as genes, transcripts and proteins (@Rosati2024). This is achieved by integrating and analyzing data derived from various omics techniques, including principal component analysis (PCA), partial least squares discriminant analysis (PLS-DA), differential expression analysis (DEA)(@Peng2024), and other relevant statistical and bioinformatics analyses to identify differentiating genes and proteins.

We are looking to find similar results as @thorne2023bile, to confirm that our results are comparable. Since they accessed the viability based partly on 'metabolic functionality' we expect to see upregulations of genes that are part of metabolic pathways in the liver, in livers with high biliary viability. Such as the glycogen production pathway. We also expect to find DEGs with functions related to bile production and/or epithelial tissue, as the measured high/low biliary viabilities are based on factors such as bile pH and capacity of epithelial bile duct cells, also known as cholangiocytes.


# Methods

Different proteomics and transcriptomic methods were used to analyse and possibly discover biomarkers. The used methods and analyses are presented in the following method section.

## Proteomics 
For the proteomics analysis, samples collected by @thorne2023bile were used. These samples consist of raw mass spectrometry data obtained from bile samples of donor livers. Samples were taken at different time points during NMP, at 30 minutes, and 150 minutes, and only for livers that ended up being transplanted at the end time point. The proteomics analysis was performed using only the 30- and 150-minute samples, as many low biliary viability samples did not end up being transplanted. In total, there are 43 samples for 150 min and 44 samples for 30 min. These samples have previously undergone protein identification and quantification. The mean percentage of missingness for the features is 42% for the 30 min samples and 56% for the 150 min samples. Samples containing more than 70% missing values were removed, which led to the removal of 2 samples for 150 min (n = 41) and 1 sample for 30 min (n = 43). Features missing in more than 4 samples in at least one of the biliary viability groups were removed. The data is normalized across samples using median centering (@dubois2022normalization), this method is used to center the data over the median of the distribution of each sample. The features were then normalized by 'variance stabilizing transformation' using 'normalize_vsn' of the DEP R-package (v.1.26.0) (@zhang2018proteome), this function uses a robust variant of the maximum-likelihood estimator (@huber2002variance). Missing values in the data were identified to mostly consist of 'missing not at random' values (MNAR)(@Heymans2022), making the missing data left-censored. Thus, imputation was performed using the 'MinDet' function of DEP (v.1.26.0) (@zhang2018proteome), which uses a deterministic minimal value approach. Here all the missing values are replaced with a minimal observed value per sample. Principal component analyses (PCA)(@alma9981496143606533) were performed on the normalized and imputed data using the mdatools (v 0.14.2) package in R (@kucheryavskiy2020mdatools). 


Potential biomarker protein evaluation was done using a confusion matrix framework to assess their viability in categorizing livers into high and low biliary viability. The tested thresholds were set in intervals of 0.01 in the range of the expression value of the corresponding potential biomarker protein, and predictive performance was done by calculating accuracy, precision, and AUC-ROC from the resulting confusion matrix. Statistical significance was assessed using Fisher’s exact test to confirm whether classification outperformed chance.

Using limma (v.3.60.6) (@ritchie2015limma) a DEA was performed on all features that passed the filtering steps. In this DEA, two distinct groups are compared to each other, the comparison consists of the samples with high biliary viability against samples with low biliary viability.  First, a linear model is fitted to each protein. The Empirical Bayes method is applied for smoothing estimates of discrete probabilities and lastly, a false discovery rate correction is applied. After determining which proteins are down/upregulated, a pathway analysis was performed. The results were visualized using e.g. volcano plots and pathview. Further literature research was performed for all DEP’s with a logFC > 2 or < -2.

The supervised discrimination method Partial Least Squares Discriminant Analysis (PLS-DA), based on PLS regression, was applied using the mdatools (version 0.14.2) R package (@kucheryavskiy2020mdatools). Normalized, imputed proteomics data was used as input with samples classified into high biliary viability and low biliary viability classes. The cross validation used was - full (leave one out). The model generated from this was analyzed with different plots that showed its parameters and explained what proteins are important and what proteins are not.


## Transcriptomics
Liver tissue biopsies, stored in the cold, were taken right before the start of the NMP. RNA from these samples was isolated using the Qiagen RNeasy Lipid Tissue Mini Kit. RNA was sequenced using the Illumina NextSeq 500. The resulting reads were trimmed with TrimGalore (v.0.6.7) (@krueger2023trim), and deduplicated with SeqKit (v.2.4.0) (@shen2024seqkit2). They were then mapped with Hisat2 (v2.2.1) (@hisat2) to the human genome h38. The counts were extracted from the resulting files using featureCounts (v.2.0.3) (@liao2014featurecounts).
MultiQC (v 1.14) (@ewels2016multiqc) was used to show a comprehensive overview of the quality of multiple FASTQ samples. This generated an HTML report that summarized the results from various QC reports. Showing the combined results across all samples. This report shows if there are potential issues such as adapter contamination or high repeat rates among other possibilities.

RNA-sequencing data was aligned to the GRCh38 (@grch38) using HISAT2 (v 2.2.1) (@kim2019hisat2). HISAT2 was used to build a splice-aware genome index, using the hisat2-build command with default parameters. This build genome index was then used to align the RNA-sequencing FASTQ files. This was done with certain parameters, -U (meaning unpaired FASTQ files), and using the --phred33 score read. The output format from this aligning was a SAM file. 
All genes with a variance of 0 were removed from the data, this is done to speed up the following analyses and tools. The data was log2 transformed and scaled using R for a PCA. A differential gene expression analysis was performed on the raw count data using the R-package DESeq2 (v.1.38.3) (@love2014deseq2). 

## Statistical methods

Differential enrichment analysis (DEA) (@Peng2024) is a method for detecting differences in protein abundance across groups, where the data gets grouped based on certain phenotypes (e.g. disease vs healthy). Resulting from the DEA are differentially expressed proteins (DEPs), which are biomarkers and can be used for constructing phenotype-specific abundance profiles. DEA in proteomics is performed on quantified, normalized, and imputed data, using one of the available tools. The selected tool, limma (@ritchie2015limma) is not designed specifically for proteomics but has proven to perform well in any setting with quantified data and is a popular option for proteomics (@Peng2024). limma uses linear fitting onto each of the features in order to find differences/correlations between conditions. To improve estimates further statistical methods can be applied to the results from limma, in this article, the Empirical Bayes method (@Smyth2004) is applied for smoothing estimates of discrete probabilities, and lastly, a false discovery rate (FDR) correction (@Aggarwal2016) is applied. The FDR correction is based on @Benjamini1995 and corrects p-values for the expected proportion of the features that are falsely found to be statistically significant, as a result of multiple testing.

PCA is used on high dimensional data. It is a method of reducing the dimensions of data. This is useful for our data since we are looking at the expression of > 2000 different genes. A PCA works by calculating a covariance matrix, linear correlations, between all of the different variables. One value in this matrix is the linear correlation between 2 variables, or genes in our case. Eigenvectors will be calculated from the variance matrix, these are the directions where the variance of the data is maximized. These values can be used to calculate the loadings, or 'strength', of the variables. These loadings, every component has loadings for all the variables, can then be projected onto the data. (@alma9981496143606533)

# Results

The results of the analyses and methods (mentioned in Materials and Methods section) to discover biomarkers will be presented in the following sections.

## DEA
The DEA on the proteomics samples in total evaluated 2255 features for the 30-min samples and 1610 features for the 150-min samples. For the 30-min samples significantly enriched expressed proteins were found. The 150-min samples contained a total of 44 DEGs (Fig.1), details on e.g. p-values and logFC per differentially abundant protein can be found in Table 1. The DEA on the transcriptomics data data revealed no significant changes in the abundance of proteins between a high and low biliary viability

```{r, fig.pos='H', out.width='75%', fig.cap="Volcano plot showing all significantly upregulated- and downregulated proteins in the comparison of high biliary viability against low biliary viability in the samples taken during NMP after 150 minutes. Proteins were deemed as significantly upregulated if the adjusted p-value was \\( \\leq 0.05 \\) and the log\\(_2\\) fold change was \\( \\geq 1.0 \\). Proteins were deemed as significantly downregulated if the adjusted p-value was \\( \\leq 0.05 \\) and the log\\(_2\\) fold change was \\( \\leq -1.0 \\).", echo=FALSE}

knitr::include_graphics("img/DEG_150min.png")
```

## PCA
Figure 2 shows that the group with low biliary viability is a bit less spread throughout the second component. Samples with a high biliary viability show to have a bit higher spread over the second component compared to the samples with low viability. 3 samples with a high biliary viability and 1 sample with a low biliary viability cause a high variance in the first component. A perfect linear separation of the 2 groups is not possible.

From the PCA that was performed on the 150-minute samples (Figure 3) it becomes clear that there appears to be more of a separation between the 2 groups, compared to Figure 2. Still, there does not appear to be any clear clustering between the groups, only with a couple of samples.


```{r, fig.pos='H', out.width='75%', fig.cap="PCA performed on proteomic samples 30 minutes after starting NMP.  Shows how the samples are spread over component 1 and component 2. Biliary viability is displayed with color.", echo=FALSE}
knitr::include_graphics("img/pca30.png")
```

```{r, fig.pos='H', out.width='75%', fig.cap="PCA performed on proteomic samples 150 minutes after starting NMP.  Shows how the samples are spread over component 1 and component 2. Biliary viability is displayed with color.", echo=FALSE}
knitr::include_graphics("img/pca150.png")
```


## PLS-DA

PLS-DA was performed on the normalized, imputed proteomics data. This data was then split into two groups, high biliary viability and low biliary viability. Figure 4 shows a comparison between the 3rd component, which contains 4.62% of the info, and Component 1, which contains 21.9% of the info. The mentioned info is the amount of each component that relates to the original data file. The reason for this comparison is that component 1 - component 2 and component 2 - component 3 showed nothing significant and no groups; therefore, there were no clear results from these comparisons. In figure 5 a clear divide is visible in the middle. 


```{r, fig.pos='H', fig.cap="Weights of each protein, the weight means how much influence it has on each of the components. The names have been hidden for readability. But each spike shows an important protein.", echo=FALSE, out.width="75%"}
knitr::include_graphics("img/weights.png")
```

```{r, fig.pos='H', out.width='75%', fig.cap="PLS-DA Score plot showing component 1 – component 3, performed on the proteomic samples at 150 min after starting NMP. Showing how the samples are laid out over components 1 and 3. Viability is shown with colour.", echo=FALSE}
knitr::include_graphics("img/PLS-DA.png")
```


## Confusion Matrix
To identify proteins and thresholds with potential as biomarkers, we filtered the dataset using a p-value cut-off of 0.05. This evaluation was performed on the 150-minute time point dataset, as no proteins-threshold combination met this minimum p-value requirement at the 30-minute time point. From the filtered 150-minute dataset, 14 unique thresholds were identified where proteins demonstrated statistically significant p-values (< 0.05)(Table 2). These protein-thresholds combinations correspond to optimal expression cut-offs used to classify samples based on protein abundance.

Among the protein-thresholds combinations meeting the significance criteria, SND1 and SNRPD1 consistently appeared at multiple thresholds. Both proteins demonstrated high classification metrics across these thresholds, indicating significant categorization performance ability. Specifically, SND1 achieved the highest area under the receiver operating characteristic curve (AUROC) of 0.95 indicating a high ability to distinguish between sample classes. It was observed at two thresholds: 13.89 and 13.79, with precision values of 0.67 and 0.75, and accuracy values of 0.85 and 0.89, respectively. The specificity for both thresholds was perfect (1.0), with zero false positive rates. Similarly, SNRPD1 was found at three thresholds (13.72, 13.62, and 13.52), reaching a maximum AUROC of 0.933. Precision ranged from 0.75 to 0.83, while accuracy improved from 0.89 to 0.93 across these thresholds. Specificity remained at 1.0 with zero false positive rates. The protein HSD17B13 was observed with a statistically significant p-value and showed moderate performance metrics, including an AUROC of 0.775, precision of 0.67, and accuracy of 0.85.

In contrast are the proteins ACE2, MMP7, and PUF60. Although they met the p-value criteria, they demonstrated notably lower classification performance. These proteins showed AUROC values below 0.7 (approximately 0.65–0.66), despite high precision values (0.93). However, their specificity was significantly lower (0.44), resulting in high false positive rates (approximately 0.56).

# Discussion/Conclusion

Protein abundances were compared in a DEA between two groups and two timepoints, with low against high biliary viability at timepoint 30 minutes and at timepoint 150 minutes. Resulting from this was the identification of 44 statistically significant and differentially enriched potential biomarker proteins ($p-adjust \le 0.05$ & $log2FC \geq 1.0 \lor log2FC \le -1.0$ ) (Table 1). These biomarkers have a logFC range of -2.60 to 2.08 for the comparison of high biliary viability against a reference of low biliary viability. This means any protein with a positive logFC is more abundant in high biliary viability livers, and a negative logFC means the protein is more abundant in low biliary viability livers. A further literature search was performed for all found proteins with a $logFC \geq 2.0 \lor logFC \le -2.0$, corresponding to a fold change of 4. This results in 5 proteins that are more abundant in low biliary viability livers, including: MMP7 (logFC:  -2.60, p-adj: 7.49e-03), MUC1 (logFC: -2.49, p-adj: 7.81e-03), PGC (logFC:  -2.32, p-adj: 3.35e-02), ACE2 (logFC: -2.23, p-adj: 7.49e-03) and ABCG5 (logFC: -2.08, p-adj: 3.53e-02), and 1 protein that is more abundant in high biliary viability livers, which is COPZ1 (logFC: 2.08, p-adj: 3.83e-02).

MMP7 (matrix metallopeptidase 7) is part of a family of enzymes responsible for the breakdown of extracellular matrix proteins (@kumara2021plasma). MMP7 has been found to contribute to the pathogenesis of bile duct epithelial injury (@Allam2024) and was also highlighted in our PLS-DA. MUC1 (Mucin 1) is a mucin family member that is associated with cell signaling and proliferation pathways and expression of MUC1 is predominantly found in cholangiocytes (@thorne2023bile). MUC1 function differs in normal cells as opposed to cancer cells, in normal cells, it acts as a lubricant but in cancer cells, overexpression of MUC1 can occur, leading to aberrant glycosylation (@chen2021muc1). MUC1 is also expressed in the liver and it has been found to be overexpressed during the carcinogenesis of hepatobiliary cancers (@kasprzak2019mucins). MUC1 was found in our PLS-DA analysis, here it was found to be a significant variable that allows for differentiating between biliary viability groups. PGC (Progastricsin) in enriched abundance plays a role in tumorigenesis of hepatocellular carcinoma (@chen2019overexpressed). ACE2 (Angiotensin-converting enzyme 2) is a receptor that is involved in allowing viral entry into host cells, including members of the SARS-CoV group (@pirola2020covid19). ABCG5 (ATP-binding cassette sub-family G member 5) is needed for transporting cholesterols into bile, @yu2002abcg5 found that expression of ABCG5 increased biliary cholesterol concentrations. Increased cholesterol concentrations can lead to cholesterol supersaturation, which can cause the formation of gallstones (@dikkers2010biliary). Noticeably all of the 5 proteins with increased abundance in low liver viability, can be linked to (liver) disease, this signifies that they are interesting candidates for further investigation as potential biomarkers. 

COPZ1 (COPI Coat Complex Subunit Zeta 1) encodes for a subunit of the cytoplasmic coatamer protein complex, this complex is involved in protein transport and autophagy (@copz1_gene). COPZ1 was found to be overexpressed in high biliary viability livers, but interestingly this COPZ1 was also found overexpressed in multiple types of cancer, including cholangiocarcinoma and liver hepatocellular cancer (@hong2023comprehensive). This protein is a prognostic marker in liver hepatocellular carcinoma and plays an essential role in maintaining the survival of a variety of tumor types (@hong2023copz1). COPZ1 was also identified as an important variable for differentiating between the biliary viability groups in our PLS-DA analysis.

Our PLS-DA analysis identified one more protein that was not found in the DEA, which is SLC6A19 (Sodium-dependent neutral amino acid transporter B(o)AT1), encoding for the system B transmembrane protein. This protein actively transports neutral amino acids across the apical membrane of epithelial cells (@slc6a19_gene). This gene and protein don’t seem to have any relation to biliary viability and mainly focus on transporting amino acids in organs such as the kidneys and intestine (@broer2008amino).

The principal component analyses revealed no clear seperation between low and high billiary viability. This could be explained by the fact that the raw data contained many NA values which had to be imputed. This imputation could have caused a lower variance between the high and low billiary viability samples.

Some of our findings of our DEA and PLS-DA overlap with what @thorne2023bile have found. @thorne2023bile found three to show significant differences between biliary viability groups; MUC5B, MUC1, and LCN2. We have been able to reinforce two of these findings, MUC1 (logFC: -2.49, p-adj: 7.81e-03) and LCN2 (p-adjust: 4.38x10-4, logFC: 1.44). LCN2 (Lipocalin-2) has been found to contribute to liver fibrosis and hypertension of the portal vein (@chen2020lcn2). MUC5B was not found to show significant differences between biliary viability groups in our research.

In Figure 4, two low biliary viability livers seemed to be misclassified as high biliary viability; this possibly suggests that these livers could have unique characteristics. This group formation could also imply that proteins driving the separation are biologically relevant, and the analysis of these livers could provide valuable insight into the mechanisms differentiating between the high and low classifications. By identifying the important variables that cause the most separation, there is a deeper understanding to be gained of what molecular drivers distinguish these groups and allow for better assessment of biliary viability. 

The DGE analysis on the transcriptomics data showed no significant differences between the high and low biliary viability. This could be due to issues with the used data. We were able to download the RNA data in 2 different file formats: paired FASTQ files and BAM files. It appeared, after running feature counts on the BAM files, that these were not aligned to a reference genome. This was very unusual, BAM-files are supposed to be mapped to a reference genome. So, the FASTQ files were used to extract the counts. A new issue showed up, the alignment rate of these FASTQ files was very low (30%). Checking the quality revealed that, at random, either the reverse or forward file for every sample contained reads with a length of 6 base pairs. These reads were not used by the aligner, which caused it to have a very low alignment rate. Seeking guidance from our teachers revealed that no one could figure out how this happened. These issues could explain why we didn’t find any significant changes between high and low variability.

MinDet was chosen as the imputation method as it is a deterministic minimal value approach, meaning all missing values are replaced with the lowest observed value per sample. This method was chosen as it ensures reproducibility because it is not a stochastic method and it is also fitting for our data, which is left-censored with many missing not-at-random (MNAR) values. MNAR values are missing values that are caused by low peptide intensity signals, meaning the intensities are below the instrumental detection limit (@shi2025imputation). However, the data also, to a lesser extent, contains missing completely at random (MCAR) values, which occur independently of peptide abundance. MAR values can occur at any intensity, unlike MNAR values and thus imputing these values using the lowest found value is not optimal. However, as there were a lot more MNAR values, it was still decided to use MinDet. Still, it should be noted that this imputation method can skew results to the left, towards lower values, which can affect outcomes.

The analysis found several proteins with statistically significant categorization performance based on p-values $\le$ 0.05, evaluated across several expression thresholds. Proteins such as SND1, SNRPD1, COPG1, and COPG2 consistently demonstrated strong classification performance, with perfect specificity and zero false positive rates. SND1 achieved the highest AUROC value, while SNRPD1 reached a close maximum AUROC value. These high AUROC values, combined with the respective high precision up to and almost maximal accuracy, indicate very high classification performance. These metrics are particularly important in biomedical applications where high specificity and low false positive rates are critical to avoid incorrect classification because of the consequences of failure. In contrast, ACE2, MMP7, and PUF60 also met the statistical significance threshold but displayed lower AUROC values. Specifically, all below 0.7, or the line of adequacy. Despite this, they exhibited high precision and moderate accuracy. However, their specificity was low and the false positive rate high. This combination suggests a potential imbalance in classification outcomes. While high precision and accuracy might indicate some use in categorization, the low AUROC implies that these proteins do not consistently perform well across the range of decision thresholds. AUROC remains a more comprehensive measure of model performance, capturing both sensitivity and specificity over all possible thresholds. Therefore, low AUROC values point to limited generalizability and shed doubt on the reliability of these proteins as classifiers in larger datasets and or other, despite their statistical significance. The appearance of multiple significant thresholds for proteins like SND1 and SNRPD1 further reinforces their potential. Their ability to classify consistently across varying cut-off values highlights a robustness that is desirable in practical applications, where expression levels can vary across individuals and experimental conditions. These results emphasize the importance of evaluating multiple performance metrics when assessing biomarker candidates. While statistical significance is a necessary starting point, it is not sufficient on its own. Metrics such as AUROC, specificity, and false positive rate provide crucial context for determining the practical utility of a given protein in classification tasks. The combination of high AUROC, consistent performance across thresholds, and perfect specificity makes proteins like SND1 and SNRPD1 strong candidates for further investigation 

In this article we have presented novel proteins that could be used as biomarkers in the differentiation between ECD-donor livers with high- and low biliary viability, which brings us closer to being able to develop biomarker profiles for biliary viability levels allowing for more structural analysis of biliary viability in ECD-donor livers, potentially enlarging the donor pool. Further research is needed to validate the relationships between these proteins of interest and biliary viability.



# Appendix
```{r, echo=FALSE}
#load("../../proteomics/paper_tables/diff_expressed_genes_table_150min.Rdata")

#knitr::kable(diff_expressed_genes_150min, format="latex")
```


Table 1. All found differentially expressed proteins in the comparison between high- and low billiary viability livers after 150 minutes of NMP, with information on certain statistics such as p-values.

```{r, echo=FALSE}
#load("../../proteomics/paper_tables/extra_rounded_significant_150_min_result.Rdata")

#knitr::kable(extra_rounded_sig_150_results, format="latex")
```
Table 2. Identified significant protein-threshold combinations in the 150-minute time point dataset.

# References